developer-docs/bolt_spec-run.md

```diff
@@ -0,0 +1,325 @@
+# BoltSpec::Run
+### Overview
+The `BoltSpec::Run` module is intended to provide a method of executing bolt in process from tests or scripts. This documentation is meant to describe the available methods and point to examples.
+
+## Configuration
+
+### Config (bolt.yaml)
+Bolt configuration data normally found in `bolt.yaml` can be passed with the `:config` key in the options hash for each method. This has the highest priority. 
+
+In order to avoid passing the same config option every time you want to use it each method will use `bolt_config` if it is available in scope and no config has been passed in the `options` hash. 
+
+### Inventory (inventory.yaml)
+Bolt configuration data normally found in `inventory.yaml` can be passed with the `:inventory` key in the options hash for each method. This has the highest priority. 
+
+In order to avoid passing the same config option every time you want to use it each method will use `bolt_inventory` if it is available in scope and no config has been passed in the `options` hash. 
+
+## Beaker compatibility
+Beaker defines an `assert` method in global scope which causes a conflict with a parameter in bolt. In order to use BoltSpec with beaker you must add `require 'bolt_spec/run'` before `beaker`. See https://tickets.puppetlabs.com/browse/BOLT-1159
+
+## Module Methods
+
+### run_task
+Run a task against specified target with given configuration
+
+**parameters**
+- `task_name`, String, *required*, Name of task to run
+- `targets`, String, *required*, String representation of target(s) (could be a comma separated list, group name, uri, etc)
+- `params`, Hash, *required*, Task parameters, keys should be strings.
+- `options`, Hash, *optional*, Default options are `config` and `inventory` with value nil. Keys should be symbols.
+
+**return**
+
+Array of result hashes
+
+**result hash**
+- `node`, String, Target name task was run on
+- `type`, String, This should always be 'task'
+- `object`, String, Task name
+- `status`, String, Task result status
+- `result`, Hash, Hash of task specific results
+
+**example**
+
+The result for a task that echoes the content of a `message` parameter with invocation `run_task('bolt_spec_example', 'localhost', 'message' => 'hi')`
```

Comment:
Making this one code block from a repl may be clearer.

Reply:
Updated.